internal/native/cgo/video.c

```diff
@@ -51,6 +53,34 @@ RK_U64 get_us()
     return (RK_U64)time.tv_sec * 1000000 + (RK_U64)time.tv_nsec / 1000; /* microseconds */
 }
 
+static void ensure_sleep_mode_disabled()
+{
+    if (!sleep_mode_available)
+    {
+        return;
+    }
+
+    int fd = open(SLEEP_MODE_FILE, O_RDWR);
+    if (fd < 0)
+    {
+        log_error("Failed to open sleep mode file: %s", strerror(errno));
+        return;
+    }
+    write(fd, "0", 1);
+    close(fd);
+
+}
+
+static void detect_sleep_mode()
+{
+    if (access(SLEEP_MODE_FILE, F_OK) != 0) {
+        sleep_mode_available = false;
+        return;
+    }
+    sleep_mode_available = true;
+    ensure_sleep_mode_disabled();
```

Comment:
i feel like this function should just do the detection of the sleep mode and the ensuring of the sleep mode being disabled should be called by the consumer of this function. I feel like this function is doing too much.